### ISSUE TITLE
Ellipsis in header of Grid web component, altough there is enough space

### ISSUE THREAD

**Issue by provi-dominik** [vaadin/flow-components#6976]
### Description

If I create the following Grid without an entry, the last column header is displayed with ellipsis:
Grid<Person> grid = new Grid<>(Person.class, false);
grid.addColumn(Person::getFirstName).setHeader("First name").setFlexGrow(1);
grid.addColumn(Person::getLastName).setHeader("Last name").setFlexGrow(0);
grid.addColumn(Person::getEmail).setHeader("Email").setFlexGrow(0);
grid.addColumn(Person::getProfession).setHeader("Profession").setFlexGrow(0).setAutoWidth(true).setFrozen(true);
![image](https://github.com/user-attachments/assets/f5434b4a-87c7-446c-8cc6-b86d89aaeb28)

If I don't set the frozen property to true, the column header is displayed without an ellipsis:
![image](https://github.com/user-attachments/assets/824968ce-d1c3-43e6-84e0-60d779f1ebfc)

### Expected outcome

The last column header should appear without an ellipsis.

### Minimal reproducible example

See the code in the description. It is based on the examples in the docs:
https://vaadin.com/docs/latest/components/grid

### Steps to reproduce

Add the Grid (see code above) to a view. The view should have enough space for the grid.

### Environment

Vaadin version(s): 24.6.0
OS: Windows

### Browsers

This issue is browser related. It only happens in Edge at 100% zoom. In other browsers this issue occours, as mentioned in the comments, at different zoom factors.

**Comment by web-padawan** [comment]
Have you tried `setFrozenToEnd(true)` instead? That should be used for frozen columns at the end of the grid.

**Comment by provi-dominik** [comment]
No, but in my application the frozen column is the first column of the grid.

**Comment by yuriy-fix** [comment]
Have you tried to add frozen columns in the beginning since it's the first column of the grid?
There is a [related note](https://github.com/vaadin/flow-components/blob/8766cdf2e3b3d048a797c2b374a51c25d3bf4237/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnBase.java#L59) in JavaDoc

**Comment by provi-dominik** [comment]
@yuriy-fix Yes, I tried it. In our real scenario, the frozen column is the first column.

**Comment by yuriy-fix** [comment]
Dear @provi-dominik, could you please provide the snippet you are using to reproduce the issue? The one mentioned in the description is not reproducing it, as confirmed in the comment above.

**Comment by provi-dominik** [comment]
Hi @tomivirkki, a member of the Vaadin support team found out that this issue is related to the browser and the zoom-factor.
These are her findings: 
**Chrome (Version 131.0.6778.265):** The column name displays without ellipsis at 100% zoom. If I adjust the zoom to values greater than 100%, the name still displays correctly. However, ellipses appear when the zoom is reduced below 80%.
**Firefox (Version 133.0.3):** The column name displays without ellipsis at 100% zoom. However, when I increase the zoom above 100%, ellipses begin to appear.
**Edge (Version 131.0.2903.146):** The column name shows an ellipsis at 100% zoom. Interestingly, increasing the zoom above 100% causes the name to display correctly.

I and our users use the Edge browser with 100% zoom.
@yuriy-fix: I think this comment helps to reproduce this issue.

**Comment by paodb** [comment]
I did the testing mentioned in the previous comment as there's an open warranty request and wasn't able to reproduce it either. The issue should be updated to say it is browser related and it only happens on Edge.

**Comment by tomivirkki** [comment]
Thanks for the clarification @provi-dominik. Confirmed that the issue is reproducible with 100% zoom level on Edge (only on Windows), and on other browsers with reduced zoom levels
